### PR TITLE
AR: pass input options to widget

### PR DESCRIPTION
**Pull request by np25071984**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #16681

User case:
```
<?=
$form->field($model, 'test', ['inputOptions' => ['placeholder' => 'placeholder#1']])->widget(
    MaskedInput::className(), [
        'name' => 'phone',
        'mask' => '999-999-9999',
        'options' => ['placeholder' => 'placeholder#2']
    ]
);
?>
```
Got input with 'placeholder#2' placeholder.
```
<?=
$form->field($model, 'test', ['inputOptions' => ['placeholder' => 'placeholder#1']])->widget(
    MaskedInput::className(), [
        'name' => 'phone',
        'mask' => '999-999-9999',
    ]
);
?>
```
Got input with 'placeholder#1' placeholder.

It is very raw revision for discussion purposes only.

### CODE REVIEW

File: framework/widgets/ActiveField.php [hunk 1]
```diff
@@ -763,6 +763,15 @@ public function radioList($items, $options = [])
      */
     public function widget($class, $config = [])
     {
+
+        if (!isset($config['options'])) {
+            $config['options'] = [];
+        }
```
Comment (rob006) [inline framework/widgets/ActiveField.php]: This looks unnecessary.

**Comment by samdark** [comment]
Any idea what's wrong with tests?

**Comment by GHopperMSK** [comment]
The original problem is located here: https://github.com/yiisoft/yii2/blob/ec1809bed2efb664ed54087e647ba9981c3c305c/framework/widgets/MaskedInput.php#L121-L128
`$this->registerClientScript()` fails due to
https://github.com/yiisoft/yii2/blob/ec1809bed2efb664ed54087e647ba9981c3c305c/framework/web/AssetManager.php#L454-L456
throws the error when `$path = '/bower-asset/inputmask/dist'`

So, if I add `var_dump(__DIR__, $path);exit;`, I will see
```
string(32) "/var/www/yii2.test/framework/web"
string(27) "/bower-asset/inputmask/dist"
```

I don't understand what is going wrong here. I need an advice.

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**